core/amber/src/main/python/core/architecture/managers/statistics_manager.py

```diff
@@ -21,11 +22,26 @@ def get_statistics(self) -> typing.Tuple[int, int]:
             - self._control_processing_time,
         )
 
-    def increase_input_tuple_count(self) -> None:
-        self._input_tuple_count += 1
-
-    def increase_output_tuple_count(self) -> None:
-        self._output_tuple_count += 1
+    def increase_input_tuple_count(self, port_id: PortIdentity) -> None:
+        if port_id is None:
+            port = 0
+        else:
+            port = port_id.id
+
+        if port not in self._input_tuple_count:
+            self._input_tuple_count[port] = 0
+        self._input_tuple_count[port] += 1
+
+    def increase_output_tuple_count(self, port_id: PortIdentity = None) -> None:
+        # Currently, the number of output ports is fixed to 1
+        if port_id is None:
+            port = 0
+        else:
+            port = port_id.id
```

Comment:
ditto. please apply the same for output count

Reply:
Fixed!